Packages/io.chainsafe.web3-unity.exchangers/Runtime/Plugins/Ramp.jslib

```diff
@@ -0,0 +1,109 @@
+            hostAppName: UTF8ToString(hostAppName),
+            userEmailAddress: UTF8ToString(userEmailAddress),
+            selectedCountryCode: UTF8ToString(selectedCountryCode),
+            defaultAsset: UTF8ToString(defaultAsset),
+            url: UTF8ToString(url),
+            webhookStatusUrl: UTF8ToString(webhookStatusUrl),
+            enabledFlows: enabledFlows,
+            offrampWebhookV3Url: UTF8ToString(offrampWebHookV3Url),
+            useSendCryptoCallback: useSendCryptoCallback
+        });
+        sdkInstance
+            .on('PURCHASE_CREATED', (event) => {
+                const purchase = event.payload.purchase;
+                // call callback
+                Module.dynCall_vidiiiiiddiiiidiidiiii(RampLib.purchaseCallback,
+                    requestId,
+                    purchase.appliedFee,
+                    stringToNewUTF8(purchase.asset.address),
+                    purchase.asset.decimals,
+                    stringToNewUTF8(purchase.asset.name),
+                    stringToNewUTF8(purchase.asset.symbol),
+                    stringToNewUTF8(purchase.asset.type),
+                    purchase.assetExchangeRate,
+                    purchase.baseRampFee,
+                    stringToNewUTF8(purchase.createdAt),
+                    stringToNewUTF8(purchase.cryptoAmount),
+                    stringToNewUTF8(purchase.endTime),
+                    // skipped escrowAddress
+                    // skipped escrowDetailsHash
+                    stringToNewUTF8(purchase.fiatCurrency),
+                    purchase.fiatValue,
+                    stringToNewUTF8(purchase.finalTxHash),
+                    stringToNewUTF8(purchase.id),
+                    purchase.networkFee,
+                    stringToNewUTF8(purchase.paymentMethodType),
+                    stringToNewUTF8(purchase.receiverAddress),
+                    stringToNewUTF8(purchase.status),
+                    stringToNewUTF8(purchase.updatedAt)
+                    );
+            })
+            .on('OFFRAMP_SALE_CREATED', (event) => {
+                const sale = event.payload.sale;
+                Module.dynCall_viiiiiiiiidi(RampLib.sellCallback,
+                    requestId,
+                    stringToNewUTF8(sale.createdAt),
+                    stringToNewUTF8(sale.crypto.amount),
+                    stringToNewUTF8(sale.crypto.assetInfo.address),
+                    stringToNewUTF8(sale.crypto.assetInfo.chain),
+                    stringToNewUTF8(sale.crypto.assetInfo.decimals),
+                    stringToNewUTF8(sale.crypto.assetInfo.name),
+                    stringToNewUTF8(sale.crypto.assetInfo.symbol),
+                    stringToNewUTF8(sale.crypto.assetInfo.type),
+                    sale.fiat.amount,
+                    stringToNewUTF8(sale.fiat.currencySymbol),
+                    );
+            })
+            .show();
+    },
+    
+    // todo use this if stringToNewUTF8() doesn't work
```

Comment:
We can probably delete this if you have tested it and it works.

Reply:
Haven't tested it yet. Can't fix the preceding code.